libraries/src/main/java/com/foodie/jwt/JwtAuthenticationFilter.java

```diff
@@ -0,0 +1,68 @@
+package com.foodie.jwt;
+
+import java.io.IOException;
+
+import org.springframework.lang.NonNull;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import com.foodie.models.repositories.TokenRepository;
+
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.RequiredArgsConstructor;
+
+@Component
+@RequiredArgsConstructor
+public class JwtAuthenticationFilter extends OncePerRequestFilter{
+
+    private final JwtService jwtService;
+    private final TokenRepository tokenRepository;
+    private final UserDetailsService userDetailsService;
+    
+    @Override
+    protected void doFilterInternal(@NonNull HttpServletRequest request, 
+        @NonNull HttpServletResponse response, 
+        @NonNull FilterChain filterChain) throws ServletException, IOException {
+
+        if (request.getServletPath().contains("/api/v1/auth")) {
+            filterChain.doFilter(request, response);
+            return;
+        }
+
+        final String jwt;
+        final String userEmail;
+        final String authHeader = request.getHeader("Authorization");
+        
+        if (authHeader == null || !authHeader.startsWith("Bearer ")) {
+            filterChain.doFilter(request, response);
+            throw new RuntimeException("Invalid JWT token");
```

Comment:
Define and throw a dedicated exception instead of using a generic one. Ideally, you would extend the RuntimeException class and then throw your custom exception Define and throw a dedicated exception instead of using a generic one. Ideally, you would extend the RuntimeException class and then throw your custom exception